### PR TITLE
Feature/devsu 1445 add airbnb plugin

### PR DESCRIPTION
**Pull request by Anna-log7**
- Added the airbnb typescript eslint plugin
- Auto-linted, then manual linted the rest of the errors

### CODE REVIEW

File: app/components/DataTable/components/GeneViewer/index.jsx [hunk 1]
```diff
@@ -78,7 +80,7 @@ const GeneViewer = (props) => {
       structuralVariantsColumnDefs[0].cellRendererParams = { link: false };
       getData();
     }
-  }, [gene, handleClose, isOpen, reportIdent, snackbar]);
+  }, [gene, handleClose, isOpen, report, snackbar]);
```
Comment (kttkjl) [inline app/components/DataTable/components/GeneViewer/index.jsx:83]: is this auto-linted to report? report might be a different object and causes this to re-run unnecessarily, could this be more specific to `report.ident`?

File: app/components/DataTable/components/SvgViewer/index.jsx [hunk 2]
```diff
@@ -11,6 +11,8 @@ import {
 import { UncontrolledReactSVGPanZoom } from 'react-svg-pan-zoom';
 import InlineSVG from 'svg-inline-react';
 import AutoSizer from 'react-virtualized/dist/es/AutoSizer';
+// not ideal, but no errors are thrown and no render loops occur
+// eslint-disable-next-line import/no-cycle
 import DataTable from '../..';
```
Comment (kttkjl) [inline app/components/DataTable/components/SvgViewer/index.jsx:16]: very scared of this

Comment (creisle) [reply app/components/DataTable/components/SvgViewer/index.jsx:16]: ya, this is a bit sketchy. @Anna-log7  might be a good thing to make a github/jira issue/ticket to fix this in a future PR unless it's something you can easily fix now

File: app/components/DataTable/index.tsx [hunk 3]
```diff
@@ -16,10 +16,13 @@ import GetAppIcon from '@material-ui/icons/GetApp';
...
 import ColumnPicker from './components/ColumnPicker';
 import LinkCellRenderer from './components/LinkCellRenderer';
+// not ideal, but no errors are thrown and no render loops occur
+// eslint-disable-next-line import/no-cycle
 import GeneCellRenderer from './components/GeneCellRenderer';
+// eslint-disable-next-line import/no-cycle
 import ActionCellRenderer from './components/ActionCellRenderer';
 import { getDate } from '../../utils/date';
-import ReportContext from '../../components/ReportContext';
+import ReportContext from '../ReportContext';
```
Comment (creisle) [inline app/components/DataTable/index.tsx:25]: to be consistent, might be worth using the `@` shorthand here despite the fact that its longer just so that you always import from `@/components` the same way when its outside the current component

File: app/components/UserAutocomplete/index.tsx [hunk 4]
```diff
@@ -7,26 +7,26 @@ import {
 } from '@material-ui/core';
 
 import api from '../../services/api';
-import { userType } from '../../common';
+import { UserType } from '../../common';
```
Comment (creisle) [inline app/components/UserAutocomplete/index.tsx:10]: These types could also use the shorthand import `@`

File: app/views/ReportView/components/GenomicSummary/index.tsx [hunk 5]
```diff
@@ -18,12 +18,12 @@ import { getComparators } from '@/services/reports/comparators';
 import { getMutationSignatures } from '@/services/reports/mutation-signature';
 import { getMutationBurden } from '@/services/reports/mutation-burden';
 import { formatDate } from '@/utils/date';
-import ReportContext from '../../../../components/ReportContext';
 import EditContext from '@/components/EditContext';
 import ConfirmContext from '@/components/ConfirmContext';
 import AlterationsService from '@/services/reports/genomic-alterations.service';
 import ReadOnlyTextField from '@/components/ReadOnlyTextField';
 import DescriptionList from '@/components/DescriptionList';
+import ReportContext from '../../../../components/ReportContext';
```
Comment (kttkjl) [inline app/views/ReportView/components/GenomicSummary/index.tsx:26]: '@' probably maybe

File: app/views/ReportView/components/MutationBurden/components/TabCards/index.tsx [hunk 6]
```diff
@@ -8,8 +8,8 @@ import {
   CardContent,
 } from '@material-ui/core';
 
-import { imageType, comparatorType, mutationBurdenType } from '../../types';
-import FrontPageTooltip from '../../../../../../components/FrontPageTooltip'; 
+import { ImageType, ComparatorType, MutationBurdenType } from '../../types';
+import FrontPageTooltip from '../../../../../../components/FrontPageTooltip';
 import Image from '../../../../../../components/Image';
```
Comment (kttkjl) [inline app/views/ReportView/components/MutationBurden/components/TabCards/index.tsx:13]: '@' boop

File: app/views/ReportView/components/ProbeSummary/index.tsx [hunk 7]
```diff
@@ -20,7 +20,6 @@ import TargetedGenesService from '../../../../services/reports/targeted-genes.se
 import TestInformationService from '../../../../services/reports/test-information.service';
 import { formatDate } from '../../../../utils/date';
 import TestInformation from './components/TestInformation';
-import { TestInformationInterface } from './components/TestInformation/interfaces';
 import SignatureCard from '../../../../components/SignatureCard';
 import PatientEdit from '../GenomicSummary/components/PatientEdit';
```
Comment (kttkjl) [inline app/views/ReportView/components/ProbeSummary/index.tsx:24]: most of these can use '@' as well ... sorry seems kinda scope creep

File: app/views/ReportView/components/TherapeuticTargets/index.tsx [hunk 8]
```diff
@@ -8,12 +8,12 @@ import { useSnackbar } from 'notistack';
 
 import DataTable from '@/components/DataTable';
 import EditContext from '@/components/EditContext';
+import api from '@/services/api';
 import DemoDescription from '@/components/DemoDescription';
 import ReportContext from '../../../../components/ReportContext';
```
Comment (kttkjl) [inline app/views/ReportView/components/TherapeuticTargets/index.tsx:13]: '@' as well ?

File: app/views/TemplateView/index.tsx [hunk 9]
```diff
@@ -4,12 +4,11 @@ import {
 } from '@material-ui/core';
 import { useSnackbar } from 'notistack';
 
+import DataTable from '@/components/DataTable';
 import api from '../../services/api';
```
Comment (kttkjl) [inline app/views/TemplateView/index.tsx:8]: '@'

File: app/views/TemplateView/components/AddEditTemplate/index.tsx [hunk 10]
```diff
@@ -20,9 +20,9 @@ import {
 import HighlightOffIcon from '@material-ui/icons/HighlightOff';
 import { useSnackbar } from 'notistack';
 
-import { recordDefaults } from '@/common';
-import sections from '../../sections';
+import { RecordDefaults } from '@/common';
 import api from '@/services/api';
+import sections from '../../sections';
```
Comment (kttkjl) [inline app/views/TemplateView/components/AddEditTemplate/index.tsx:25]: '@'

File: app/components/DataTable/components/SvgViewer/index.jsx (hunk 2 above)
Comment (Anna-log7) [reply app/components/DataTable/components/SvgViewer/index.jsx:16]: I'm open to suggestions on how to fix this, but I thought about it for a bit and the only thing I could come up with was using a new AgGridReact table which isn't DataTable in here

Comment (Anna-log7) [reply app/components/DataTable/components/SvgViewer/index.jsx:16]: This issue also affects the `GeneViewer` component to `DataTable`

Comment (creisle) [reply app/components/DataTable/components/SvgViewer/index.jsx:16]: I think just making a ticket detailing the problem is sufficient for this PR since its more about the linting, you can work on fixing the circular imports now that they are highlighted in a subsequent one. 

I know I had a similar problem for graphkb client at one point and I fixed it by splitting the bulk of the component out and making 2 top-level components that used it. one imported the other

Comment (Anna-log7) [reply app/components/DataTable/components/SvgViewer/index.jsx]: I just ended up fixing it as I described above

File: app/components/DataTable/components/GeneViewer/index.jsx [hunk 11]
```diff
@@ -39,22 +37,13 @@ const GeneViewer = (props) => {
   const {
     onClose,
     isOpen,
+    reportIdent,
     gene,
   } = props;
 
-  const { report } = useContext(ReportContext);
```
Comment (kttkjl) [inline app/components/DataTable/components/GeneViewer/index.jsx]: was this causing problems ...? 

if you're using `reportIdent` btw, prop-types is complaining

Comment (Anna-log7) [reply app/components/DataTable/components/GeneViewer/index.jsx]: Some of the `DataTable`s are rendered within angular which can't access the react context

Comment (kttkjl) [reply app/components/DataTable/components/GeneViewer/index.jsx]: is /report/61ec99d8-557f-44e7-aa00-27e2e5442bb5/small-mutations working for you when you click on a gene? seems like if we use both reportContext or reportIdent it comes back as null

Comment (creisle) [reply app/components/DataTable/components/GeneViewer/index.jsx]: I am seeing that issue as well

Comment (creisle) [reply app/components/DataTable/components/GeneViewer/index.jsx]: image

Comment (Anna-log7) [reply app/components/DataTable/components/GeneViewer/index.jsx]: Fixed with latest commit

**Review by creisle — DISMISSED** [review]

**Review by creisle — APPROVED** [review]
